src/sql/src/parsers/tql_parser.rs

```diff
@@ -0,0 +1,205 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use snafu::ResultExt;
+use sqlparser::keywords::Keyword;
+use sqlparser::parser::ParserError;
+use sqlparser::tokenizer::Token;
+
+use crate::error::{self, Result};
+use crate::parser::ParserContext;
+use crate::statements::statement::Statement;
+use crate::statements::tql::{Tql, TqlEval, TqlExplain};
+
+pub const TQL: &str = "TQL";
+const EVAL: &str = "EVAL";
+const EXPLAIN: &str = "EXPLAIN";
+use sqlparser::parser::Parser;
+
+/// TQL extension parser, including:
+/// - TQL EVAL <query>
+/// - TQL EXPLAIN <query>
+impl<'a> ParserContext<'a> {
+    pub(crate) fn parse_tql(&mut self) -> Result<Statement> {
+        self.parser.next_token();
+
+        match self.parser.peek_token() {
+            Token::Word(w) => match w.keyword {
+                Keyword::NoKeyword if w.value.to_uppercase() == EVAL && w.quote_style.is_none() => {
```

Comment:
Do we need to support `EVALUATE` as well? Like `DESC` and `DESCRIBE`

Reply:
Agree, I'll add it.